### DISCUSSION TITLE
Feature Request: Decouple I²S microphone support from the M5 build flags

### DISCUSSION THREAD

**Discussion by FHeilmann** [PlummersSoftwareLLC/NightDriverStrip#26] [Ideas]
Happy to be proven wrong, but it seems like currently, I²S microphone support is only compiled in when `M5STICKC` or `M5STICKCPLUS` are defined, and defaults to ADC sampling via the I²S peripheral otherwise.

Are there any plans to generalize this so that users with other ESP32 boards and I²S microphones will be able to use NightDriver?

Cheers,

Florian

**Comment by davepl** [comment]
There are actually two codebases in there. If you ENABLE_AUDIO to 1, on an M5 it will use I2S to read from the M5.

 

If you ENABLE_AUDIO but don’t use an M5, I’ve used a MAX9814 equally well.  It’s a separate I2S codepath.

 

If you want REALLY old versions, like pulling the samples with digitalRead or IRQ based solutions, look at my own set of repos for things with the name SoundFrame*, where you should find some old examples.

 

I also have a complete code walkthough of the older codebase on the YouTube channel!

 

_ Dave

 

From: Florian Heilmann ***@***.***> 
Sent: Wednesday, October 6, 2021 5:20 AM
To: PlummersSoftwareLLC/NightDriverStrip ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [PlummersSoftwareLLC/NightDriverStrip] Feature Request: Decouple I²S microphone support from the M5 build flags (Discussion #26)

 

Happy to be proven wrong, but it seems like currently, I²S microphone support is only compiled in when M5STICKC or M5STICKCPLUS are defined, and defaults to ADC sampling via the I²S peripheral otherwise.

Are there any plans to generalize this so that users with other ESP32 boards and I²S microphones will be able to use NightDriver?

Cheers,

Florian

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub <https://github.com/PlummersSoftwareLLC/NightDriverStrip/discussions/26> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AA4HCFYGHCAOGXODMTQNOY3UFQ5FRANCNFSM5FONSTRQ> .
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>  or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub> .   <https://github.com/notifications/beacon/AA4HCF4RCRVAI7EWEHMXWM3UFQ5FRA5CNFSM5FONSTR2YY3PNVWWK3TUL52HS4DFVJCGS43DOVZXG2LPN2VGG33NNVSW45C7NFSM4ABXDBAQ.gif>

**Comment by FHeilmann** [comment]
I don't want to use an M5, but use an I²S microphone like the INMP441 or the SPH0645. That's why I'm asking :)
Right now the only way to get the code path that reads from an I²S microphone is to enable the M5, what I'm after is to decouple that so I can use an I²S microphone without using an M5

**Comment by stretch911** [comment]
any updates on this? I don't see build flags for I2S...i want to use a INMP441
